### ISSUE TITLE
Transport endpoint is not connected

### ISSUE THREAD

**Issue by ANaumann85** [containers/fuse-overlayfs#169]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

podman run leads to the error message __Transport endpoint is not connected__ after some commands.

**Steps to reproduce the issue:**

1. build an image (with base image debian:jessie-slim) and install  [EasyBuild](https://github.com/easybuilders/easybuild)

2. run podman with the image and command ``bash -l``. Inside the container run
````
module load EasyBuild
eb xorg-macros-1.19.2-GCCcore-7.3.0.eb -r
````
It stops after the configuration. At the moment, I cannot reduce the steps for reproduction. The error occurs only with fuse-overlayfs. I did not try without fuse (and as root).

3. run a simple command, like ls

**Describe the results you received:**

The installation of the easybuild configuration breaks. After the last command, I got the error message __Transport endpoint is not connected__

**Describe the results you expected:**
no error message

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:            1.7.0-dev
RemoteAPI Version:  1
Go Version:         go1.11.6
Git Commit:         fcd48db4d24f6dba4fb2652d72aa0d86e167aa0c
Built:              Sun Dec 22 22:59:44 2019
OS/Arch:            linux/amd64

```

**Output of `podman info --debug`:**

```
debug:
  compiler: gc
  git commit: fcd48db4d24f6dba4fb2652d72aa0d86e167aa0c
  go version: go1.11.6
  podman version: 1.7.0-dev
host:
  BuildahVersion: 1.12.0
  CgroupVersion: v1
  Conmon:
    package: Unknown
    path: /usr/local/bin/conmon
    version: 'conmon version 2.0.9-dev, commit: b17d81bd7dbfc39ea85593e2a113c0c604d6eb1b'
  Distribution:
    distribution: debian
    version: "10"
  IDMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  MemFree: 2111410176
  MemTotal: 16454664192
  OCIRuntime:
    name: runc
    package: 'runc: /usr/sbin/runc'
    path: /usr/sbin/runc
    version: |-
      runc version 1.0.0~rc6+dfsg1
      commit: 1.0.0~rc6+dfsg1-3
      spec: 1.0.1
  SwapFree: 0
  SwapTotal: 0
  arch: amd64
  cpus: 8
  eventlogger: journald
  hostname: debian-X1
  kernel: 4.19.0-6-amd64
  os: linux
  rootless: true
  uptime: 228h 5m 45.95s (Approximately 9.50 days)
registries:
  search:
  - docker.io
  - registry.fedoraproject.org
  - registry.access.redhat.com
store:
  ConfigFile: /home/andreas/.config/containers/storage.conf
  ContainerStore:
    number: 1
  GraphDriverName: overlay
  GraphOptions:
    overlay.mount_program:
      Executable: /usr/local/bin/fuse-overlayfs
      Package: Unknown
      Version: |-
        fusermount3 version: 3.4.1
        fuse-overlayfs: version 0.7.3
        FUSE library version 3.4.1
        using FUSE kernel interface version 7.27
    overlay.mountopt: ro
  GraphRoot: /home/andreas/.local/share/containers/storage
  GraphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  ImageStore:
    number: 2
  RunRoot: /run/user/1000/containers
  VolumePath: /home/andreas/.local/share/containers/storage/volumes

```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**
installed from source

**Additional environment details (AWS, VirtualBox, physical, etc.):**

- physical machine
- distribution: Debian GNU/Linux 10

**Comment by ANaumann85** [comment]
I was not able to reduce the error to a minmal container with only a Dockerfile. Therefore, I provide the archive [transport_endpoint_not_connected.tar.gz](https://github.com/containers/fuse-overlayfs/files/4061245/transport_endpoint_not_connected.tar.gz), which contains :
* a README.md, which states the commands and my output
* a Dockerfile
* the script installPre.sh it installs modules-environment and easybuild. 
* the script produceError.sh It tries to install an easybuild module, which fails and leads to the error 
* the python script bootstrap_eb.py and archive modules.tar.gz are used by the install script above.

What I see from the python trace is:
1. easybuild starts build and logging to a temporary folder in /tmp/
2. Some times later, it checks again for the existence of the temporary folder, which fails (even that is questionable)
3. In turn it tries to create the non-existing folder, which fails with the error "folder exists" 

So there is something really odd.

**Comment by giuseppe** [comment]
thanks for the reproducer, it was very helpful to find out the root cause for the isue.

I've opened a PR here: https://github.com/containers/fuse-overlayfs/pull/170

Could you check if it solves the issue for you too?

**Comment by ANaumann85** [comment]
Yes, it did solve the problem for me. That was quite fast, thank you!

**Comment by ANaumann85** [comment]
It looks like I am loosing files now. Is there some kind of logging, which file was changed or removed? The new test takes around one hour, so I would try to simplify the test case.

**Comment by giuseppe** [comment]
No that is not logged, what file are you losing?  So I can take a look locally at what is happening

**Comment by giuseppe** [comment]
> It looks like I am loosing files now

I've tried the same container both as rootless (with fuse-overlayfs) and as root, and with both I get the same:

```
$ find /home | wc -l
18226
```

**Comment by ANaumann85** [comment]
I created another [script](https://github.com/containers/fuse-overlayfs/files/4064815/produceError.sh.txt), which runs on the same container as in [the previous reproducer](https://github.com/containers/fuse-overlayfs/issues/169#issuecomment-574351011).
(github does not allow script uploads, hence I added the suffix .txt)

But now it takes around 30 minutes to complete. It first shows the content of a directory, then tries to install some easybuild modules and afterwards it shows the content of the same directory again.
